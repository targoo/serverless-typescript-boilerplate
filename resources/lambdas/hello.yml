hello-status:
  handler: src/modules/hello/status.handler
  iamRoleStatementsName: ${self:custom.stackname}-hello-status
  iamRoleStatements:
    - Effect: 'Allow'
      Action:
        - dynamodb:GetItem
      Resource: 'arn:aws:dynamodb:${self:provider.region}:*:table/${self:custom.stackname}-main'
  timeout: 29
  memorySize: 128
  reservedConcurrency: 5
  events:
    - http:
        method: GET
        path: /hello
  package:
    include:
      - src/hello/**
