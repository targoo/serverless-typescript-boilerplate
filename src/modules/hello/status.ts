// eslint-disable-next-line no-unused-vars
import { APIGatewayEvent, Context, Handler, Callback } from 'aws-lambda';

import { createResponse } from '../../utils/response';

export const handler: Handler = async (event: APIGatewayEvent, _context: Context, callback: Callback) => {
  const response = createResponse(200, {
    message: {
      message: 'Go Serverless v2.0! Your function executed successfully!',
      input: event,
    },
  });

  callback(null, response);
};
