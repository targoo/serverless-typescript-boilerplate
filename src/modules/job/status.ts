import { successResponse } from '../../utils/lambda-response';

export const handler = async (event, _context, callback) => {
  const { requestContext: { identity: { cognitoAuthenticationProvider = '' } = {} } = {} } = event;

  // Cognito authentication provider looks like:
  // cognito-idp.eu-west-1.amazonaws.com/eu-west-1_hZKjWiSMb,cognito-idp.eu-west-1.amazonaws.com/eu-west-1_hZKjWiSMb:CognitoSignIn:8e978455-86f9-4a18-bcd4-e307115961ca
  // Where us-east-1_aaaaaaaaa is the User Pool id
  // And qqqqqqqq-1111-2222-3333-rrrrrrrrrrrr is the User Pool User Id
  // https://serverless-stack.com/chapters/mapping-cognito-identity-id-and-user-pool-id.html
  const userId = cognitoAuthenticationProvider.split(':').pop();

  const identity = {
    userId,
  };

  const response = successResponse({ identity: identity });

  callback(null, response);
};
