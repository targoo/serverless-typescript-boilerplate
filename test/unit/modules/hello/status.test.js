import { handler } from '../../../../src/modules/hello/status';

describe('handler', () => {
  it('executes as expected', () => {
    const cb = jest.fn();
    handler({}, {}, cb);
    expect(cb).toHaveBeenCalled();
    expect(cb).toMatchSnapshot();
  });
});
