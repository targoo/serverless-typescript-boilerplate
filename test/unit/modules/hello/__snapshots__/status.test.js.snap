// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handler executes as expected 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      null,
      Object {
        "body": "{\\"message\\":{\\"message\\":\\"Go Serverless v2.0! Your function executed successfully!\\",\\"input\\":{}}}",
        "headers": Object {
          "Access-Control-Allow-Origin": "*",
        },
        "statusCode": 200,
      },
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
